packages/next/src/build/webpack-config.ts

```diff
@@ -387,35 +397,64 @@ function getReactProfilingInProduction() {
-    'react-server-dom-webpack/server.node$': `next/dist/compiled/react-server-dom-webpack${bundledReactChannel}/server.node`,
+  let alias: Record<string, string> = {}
+  if (opts.layer === 'app-pages-browser' || opts.isEdgeServer) {
+    alias = {
+      react$: `next/dist/compiled/react${bundledReactChannel}`,
+      'react-dom$': `next/dist/compiled/react-dom${bundledReactChannel}`,
+      'react/jsx-runtime$': `next/dist/compiled/react${bundledReactChannel}/jsx-runtime`,
+      'react/jsx-dev-runtime$': `next/dist/compiled/react${bundledReactChannel}/jsx-dev-runtime`,
+      'react-dom/client$': `next/dist/compiled/react-dom${bundledReactChannel}/client`,
+      'react-dom/server$': `next/dist/compiled/react-dom${bundledReactChannel}/server`,
+      'react-dom/server.edge$': `next/dist/compiled/react-dom${bundledReactChannel}/server.edge`,
+      'react-dom/server.browser$': `next/dist/compiled/react-dom${bundledReactChannel}/server.browser`,
+      'react-server-dom-webpack/client$': `next/dist/compiled/react-server-dom-webpack${bundledReactChannel}/client`,
+      'react-server-dom-webpack/client.edge$': `next/dist/compiled/react-server-dom-webpack${bundledReactChannel}/client.edge`,
+      'react-server-dom-webpack/server.edge$': `next/dist/compiled/react-server-dom-webpack${bundledReactChannel}/server.edge`,
+      'react-server-dom-webpack/server.node$': `next/dist/compiled/react-server-dom-webpack${bundledReactChannel}/server.node`,
+    }
+  } else if (opts.layer === 'ssr') {
+    alias = {
+      'react/jsx-runtime$': `next/dist/server/future/route-modules/app-page/vendored/shared/react-jsx-runtime`,
+      'react/jsx-dev-runtime$': `next/dist/server/future/route-modules/app-page/vendored/shared/react-jsx-dev-runtime`,
+      react$: `next/dist/server/future/route-modules/app-page/vendored/${opts.layer}/react`,
+      'react-dom$': `next/dist/server/future/route-modules/app-page/vendored/${opts.layer}/react-dom`,
+      'react-dom/server.edge$': `next/dist/server/future/route-modules/app-page/vendored/${opts.layer}/react-dom-server-edge`,
+      'react-server-dom-webpack/client.edge$': `next/dist/server/future/route-modules/app-page/vendored/${opts.layer}/react-server-dom-webpack-client-edge`,
+      // not essential but we're providing this alias for people who might use it
+      'react-dom/server$': `next/dist/server/future/route-modules/app-page/vendored/${opts.layer}/react-dom-server-edge`,
+    }
+  } else if (opts.layer === 'rsc') {
+    alias = {
+      'react/jsx-runtime$': `next/dist/server/future/route-modules/app-page/vendored/shared/react-jsx-runtime`,
+      'react/jsx-dev-runtime$': `next/dist/server/future/route-modules/app-page/vendored/shared/react-jsx-dev-runtime`,
+      react$: `next/dist/server/future/route-modules/app-page/vendored/${opts.layer}/react`,
+      'react-dom$': `next/dist/server/future/route-modules/app-page/vendored/${opts.layer}/react-dom`,
```

Comment:
Does this work with production profiling? I suppose if you alias to the production profiler then it will use the distributed package not the runtime package but then that package should get linked up to the runtime for it's dependencies.

Also production profiling only makes sense for browser builds so it probably should be gated by layer too. I can't leave a comment on those lines directly because they werne't changed in this PR but look at line 476 to see what I mean.

Reply:
As discussed offline, I'll check profiling manually + add a test later on. I think it works though